internal/querynodev2/delegator/segment_pruner.go

```diff
@@ -102,7 +102,11 @@ func PruneSegments(ctx context.Context,
 		}
 
 		// 1. parse expr for prune
-		expr := ParseExpr(exprPb, NewParseContext(clusteringKeyField.GetFieldID(), clusteringKeyField.GetDataType()))
+		expr, err := ParseExpr(exprPb, NewParseContext(clusteringKeyField.GetFieldID(), clusteringKeyField.GetDataType()))
+		if err != nil {
+			log.Ctx(ctx).RatedWarn(10, "failed to parse expr for segment prune, fallback to common search/query", zap.Error(err))
+			return
+		}
 
```

Comment:
if parse expr for pruner failed, directly skip prune process and fall back to brute-force query/search